### PR TITLE
Add a new translation to deal with Docker rate limits.

### PR DESCRIPTION
**Pull request by mfunaki**
# Description
A new Japanese translation for 'Setting Up Docker Hub Pull Through Mirror'

# Reasons
English version has recently added to explain how to deal with Docker rate limits (for CircleCI Server).
This is a translation for that document.

### CODE REVIEW

File: jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc [hunk 1]
```diff
@@ -0,0 +1,76 @@
+---
+version:
+- Server v2.x
+- Server Admin
+---
+= Docker Hub のプル スルー ミラーの設定
```
Comment (ganezasan) [inline jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc:6]:
```suggestion
= Docker Hub のミラーの使用
```

File: jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+:page-layout: classic-docs
+:page-liquid:
+:icons: font
+:toc: macro
+:toc-title:
+
+ここでは、Docker Hub のプル スルー レジストリ ミラーを設定する方法について説明します。
```
Comment (ganezasan) [inline jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc:13]:
```suggestion
ここでは、Docker Hub のレジストリのミラーを設定する方法について説明します。
```

File: jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
++
+例として、このサーバーの IP アドレスを `192.0.2.1` とします。
+
+3. 以下のコマンドを実行して、*キャッシュ サーバーを起動* します。
++
+NOTE: `DOCKER_HUB_USERNAME` は Docker Hub アカウントのユーザー名に、`DOCKER_HUB_ACCESS_TOKEN` は https://hub.docker.com/settings/security で取得したアクセス トークンに置き換えてください。
++
+`sudo docker run -d -p 80:5000 --restart=always --name=through-cache -e REGISTRY_PROXY_REMOTEURL="https://registry-1.docker.io" -e REGISTRY_PROXY_USERNAME=DOCKER_HUB_USERNAME -e REGISTRY_PROXY_PASSWORD=DOCKER_HUB_ACCESS_TOKEN registry`
+
+4. TCP ポート 80 (HTTP) が開いていてアクセス可能であることを確認します。セキュリティを強化するには、Nomad クライアントのみにポートを開放することをお勧めします。
```
Comment (makotom) [inline jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc:43]:
```suggestion
4. TCP ポート 80 (HTTP) が開いていてアクセス可能であることを確認します。セキュリティ強化のため、Nomad クライアントのみにポートを開放することをお勧めします。
```

File: jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
++
+NOTE: `192.0.2.1` は実際のプル スルー キャッシュ サーバーの IP アドレスに置き換えてください。
++
+`sudo bash -c 'cat <<< $(jq ".¥"registry-mirrors¥" = [¥"http://192.0.2.1¥"]" /etc/docker/daemon.json) > /etc/docker/daemon.json'`
+
+2. *Docker デーモンをリロード* して、設定を適用します。
++
+`sudo systemctl restart docker.service`
+
+これで、Docker Hub からキャッシュ サーバー経由で Docker イメージがダウンロードされることを確認できるはずです。副次的な効果として、キャッシュ サーバーに構成されているユーザーのプライベート イメージが、明示的な認証なしにダウンロードできます。
```
Comment (makotom) [inline jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc:59]:
```suggestion
これで、Docker Hub からキャッシュ サーバー経由で Docker イメージがダウンロードされることを確認できるはずです。副次的な効果として、キャッシュ サーバーで構成されているユーザーのプライベート イメージが、明示的な認証なしにダウンロードできます。
```

File: jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+:toc: macro
+:toc-title:
+
+ここでは、Docker Hub のプル スルー レジストリ ミラーを設定する方法について説明します。
+
+toc::[]
+
+== 概要
+
+https://www.docker.com/blog/scaling-docker-to-serve-millions-more-developers-network-egress/[2020 年 11 月 1 日]から、Docker Hub で匿名ユーザーに対する発信元 IP に基づいたプル回数制限が導入されます。運用中のサービスでの混乱を避けるために、Docker Hub のプルを認証するようにすることをお勧めします。認証は設定ファイルで行えます (https://circleci.com/ja/docs/2.0/private-images/[Docker の認証付きプルの使用]を参照)。
```
Comment (ganezasan) [inline jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc:19]:
```suggestion
(https://www.docker.com/blog/scaling-docker-to-serve-millions-more-developers-network-egress/)[2020年11月1 日]から、Docker Hub で匿名ユーザーに対する発信元 IP に基づいたプル回数制限が導入されます。運用中のサービスでの混乱を避けるために、Docker Hub から認証付きでプル操作を行うようにすることをお勧めします。認証は設定ファイルで行えます (https://circleci.com/ja/docs/2.0/private-images/)[Docker の認証付きプルの使用]を参照。
```

File: jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+あるいは、Docker Hub アカウントの資格情報をあらかじめ構成した Docker Hub プル スルー レジストリ ミラーをセットアップするという方法もあります。プル スルー レジストリ ミラーを使用すると、場合によっては多数の設定ファイルを変更するよりも簡単に対応できます。また、Docker Hub へのネットワーク ラウンドトリップが減るため、パフォーマンスがさらに向上する可能性もあります。
```
Comment (ganezasan) [inline jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc:21]:
```suggestion
あるいは、Docker Hub アカウントの資格情報をあらかじめ構成した Docker Hub のレジストリのミラーをセットアップするという方法もあります。レジストリのミラーを使用すると、場合によっては多数の設定ファイルを変更するよりも簡単に対応できます。また、Docker Hub への通信が減るため、パフォーマンスがさらに向上する可能性もあります。
```

File: jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+この設定手順は大きく 2 段階に分かれます。まずプル スルー レジストリ ミラーをセットアップし、次に Nomad クライアントでミラーを使用するように設定します。
```
Comment (ganezasan) [inline jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc:23]:
```suggestion
この設定手順は大きく 2 段階に分かれます。まずレジストリのミラーをセットアップし、次に Nomad クライアントでミラーを使用するように設定します。
```

File: jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+== プル スルー キャッシュ レジストリ (Docker Hub のプロキシ) をセットアップする
```
Comment (ganezasan) [inline jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc:25]:
```suggestion
== レジストリのミラー(Docker Hub のプロキシ) をセットアップする
```

File: jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+
+1. *Docker Hub アカウントを用意* します。これは、プル スルー キャッシュ レジストリから Docker Hub にアクセスする際に使用します。
```
Comment (ganezasan) [inline jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc:27]:
```suggestion
1. *Docker Hub アカウントを用意* します。これは、レジストリのミラーから Docker Hub にアクセスする際に使用します。
```

File: jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
++
+NOTE: `DOCKER_HUB_USERNAME` は Docker Hub アカウントのユーザー名に、`DOCKER_HUB_ACCESS_TOKEN` は https://hub.docker.com/settings/security で取得したアクセス トークンに置き換えてください。
++
+`sudo docker run -d -p 80:5000 --restart=always --name=through-cache -e REGISTRY_PROXY_REMOTEURL="https://registry-1.docker.io" -e REGISTRY_PROXY_USERNAME=DOCKER_HUB_USERNAME -e REGISTRY_PROXY_PASSWORD=DOCKER_HUB_ACCESS_TOKEN registry`
+
+4. TCP ポート 80 (HTTP) が開いていてアクセス可能であることを確認します。セキュリティを強化するには、Nomad クライアントのみにポートを開放することをお勧めします。
++
+たとえば AWS では、セキュリティ グループと OS レベルのファイアウォールの両方が HTTP 経由で Nomad クライアントからの接続を受け付けるように正しく設定されていることを確認する必要があります。
+
+== Nomad クライアントがプル スルー キャッシュ サーバーを使用するように設定する (Nomad クライアント「ごと」に実行)
```
Comment (ganezasan) [inline jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc:47]:
```suggestion
== Nomad クライアントがミラーのレジストリを使用するように設定する (Nomad クライアント「ごと」に実行)
```

File: jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc (hunk 4 above)
Comment (ganezasan) [inline jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc:59]:
```suggestion
これで、Docker Hub からミラーのレジストリ経由で Docker イメージがダウンロードされることを確認できるはずです。副次的な効果として、ミラーのレジストリに構成されているユーザーのプライベート イメージが、明示的な認証なしにダウンロードできます。
```

File: jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+
+Nomad クライアント「ごと」に以下の手順を実行します。
+
+1. *`/etc/docker/daemon.json` の `registry-mirrors` オプションを削除* します。ファイルの編集後も、*`/etc/docker/daemon.json` が有効な JSON ファイルであることを確認* してください。
+
+2. `sudo systemctl restart docker.service` を実行して、変更を適用します。
+
+== 関連資料
+
+* https://docs.docker.com/registry/recipes/mirror/ (プル スルー キャッシュ サーバー
```
Comment (ganezasan) [inline jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc:71]:
```suggestion
* https://docs.docker.com/registry/recipes/mirror/ (ミラーのレジストリ

**Review by ganezasan — CHANGES_REQUESTED** [review]
大きな変更でプルスルーを削除し、レジストリのミラーという表現に変更しました。

**Review by ganezasan — APPROVED** [review]

File: jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+
+Nomad クライアント「ごと」に以下の手順を実行します。
+
+1. *`/etc/docker/daemon.json` の `registry-mirrors` オプションを削除* します。ファイルの編集後も、*`/etc/docker/daemon.json` が有効な JSON ファイルであることを確認* してください。
+
+2. `sudo systemctl restart docker.service` を実行して、変更を適用します。
+
+== 関連資料
+
+* https://docs.docker.com/registry/recipes/mirror/ (ミラーのレジストリ
```
Comment (ganezasan) [inline jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc:71]:
```suggestion
* https://docs.docker.com/registry/recipes/mirror/ (レジストリのミラー
```

File: jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
++
+NOTE: `DOCKER_HUB_USERNAME` は Docker Hub アカウントのユーザー名に、`DOCKER_HUB_ACCESS_TOKEN` は https://hub.docker.com/settings/security で取得したアクセス トークンに置き換えてください。
++
+`sudo docker run -d -p 80:5000 --restart=always --name=through-cache -e REGISTRY_PROXY_REMOTEURL="https://registry-1.docker.io" -e REGISTRY_PROXY_USERNAME=DOCKER_HUB_USERNAME -e REGISTRY_PROXY_PASSWORD=DOCKER_HUB_ACCESS_TOKEN registry`
+
+4. TCP ポート 80 (HTTP) が開いていてアクセス可能であることを確認します。セキュリティ強化のため、Nomad クライアントのみにポートを開放することをお勧めします。
++
+たとえば AWS では、セキュリティ グループと OS レベルのファイアウォールの両方が HTTP 経由で Nomad クライアントからの接続を受け付けるように正しく設定されていることを確認する必要があります。
+
+== Nomad クライアントがミラーのレジストリを使用するように設定する (Nomad クライアント「ごと」に実行)
```
Comment (ganezasan) [inline jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc:47]:
```suggestion
== Nomad クライアントがレジストリのミラーを使用するように設定する (Nomad クライアント「ごと」に実行)
```

File: jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
++
+NOTE: `192.0.2.1` は実際のプル スルー キャッシュ サーバーの IP アドレスに置き換えてください。
++
+`sudo bash -c 'cat <<< $(jq ".¥"registry-mirrors¥" = [¥"http://192.0.2.1¥"]" /etc/docker/daemon.json) > /etc/docker/daemon.json'`
+
+2. *Docker デーモンをリロード* して、設定を適用します。
++
+`sudo systemctl restart docker.service`
+
+これで、Docker Hub からキャッシュ サーバー経由で Docker イメージがダウンロードされることを確認できるはずです。副次的な効果として、キャッシュ サーバーで構成されているユーザーのプライベート イメージが、明示的な認証なしにダウンロードできます。
```
Comment (ganezasan) [inline jekyll/_cci2_ja/docker-hub-pull-through-mirror.adoc:59]:
```suggestion
これで、Docker Hub からレジストリのミラー経由で Docker イメージがダウンロードされることを確認できるはずです。副次的な効果として、レジストリのミラーにキャッシュされているユーザーのプライベート イメージが、明示的な認証なしにダウンロードできます。
```

**Pull request merged**